org.eclipse.gef/src/org/eclipse/gef/internal/ui/palette/editparts/DrawerFigure.java

```diff
@@ -143,8 +146,26 @@ public void paintBackground(IFigure figure, Graphics g, Insets insets) {
 	 *                used to display the drawer header with an EditPartTipHelper,
 	 *                if the header is not completely visible). It can be
 	 *                <code>null</code> (the tip won't be displayed).
+	 * @deprecated Use {@link #DrawerFigure(Control, ColorPalette)} instead to
+	 *             support custom color palettes.
 	 */
+	@Deprecated(forRemoval = true, since = "3.20")
 	public DrawerFigure(final Control control) {
```

Comment:
I wonder whether it's a good or a bad idea to keep those old constructors. They're internal, so we're free to remove them without breaking API. But I know of at least one project that is using those internal classes which might expect this constructor...

Reply:
Keeping technical dept for the sake of what-ifs is probably not a good idea... Especially for something that's very easy to adapt, if user were to use those internal classes.

Reply:
I totally agree here with you. We have enough technical dept already.